### PR TITLE
Stateful ErrorSinks can now react to compilation ending.

### PR DESCRIPTION
**Pull request by maxhaton**

### CODE REVIEW

**Comment by maxhaton** [comment]
Purpose is to properly support @royalpinto007 SARIF work e.g. a valid compilation must still produce some sarif output, better to do this with an errorsink than bunch of spaghetti

**Comment by maxhaton** [comment]
@thewilsonator

**Comment by maxhaton** [comment]
the name isn't perfect (is yours?)

File: compiler/src/dmd/globals.d [hunk 1]
```diff
@@ -467,6 +467,17 @@ extern (C++) struct Global
         return major * 1000 + minor;
     }
 
+    /**
+     * Indicate to stateful error sinks that no more errors can be produced.
+     * This is to support error sinks that collect information to produce a
+     * single (say) report.
+     */
+    extern(C++) void plugErrorSinks()
```
Comment (thewilsonator) [inline compiler/src/dmd/globals.d:475]: update headers, `frontend.h` and `globals.h`

Comment (maxhaton) [reply compiler/src/dmd/globals.d:475]: done

**Comment by maxhaton** [comment]
@ibuclaw not sure if you actually need this at all for gdc but tried to make it easy if you do.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**